src/components/tree/phyloTree/helpers.js

```diff
@@ -78,27 +81,76 @@ function _getSpaceBetweenSubtrees(numSubtrees, numTips) {
  * rectangularLayout, radialLayout, createChildrenAndParents
  * side effects: <phyloNode>.displayOrder (i.e. in the redux node) and <phyloNode>.displayOrderRange
  * @param {Array<PhyloNode>} nodes
+ * @param {boolean} focus
  * @returns {undefined}
  */
-export const setDisplayOrder = (nodes) => {
+export const setDisplayOrder = (nodes, focus) => {
+  timerStart("setDisplayOrder");
+
+  const getDisplayOrder = getDisplayOrderCallback(nodes, focus);
   const numSubtrees = nodes[0].n.children.filter((n) => n.fullTipCount!==0).length;
   const numTips = nodes[0].n.fullTipCount;
   const spaceBetweenSubtrees = _getSpaceBetweenSubtrees(numSubtrees, numTips);
   let yCounter = 0;
   /* iterate through each subtree, and add padding between each */
   for (const subtree of nodes[0].n.children) {
     if (subtree.fullTipCount===0) { // don't use screen space for this subtree
-      setDisplayOrderRecursively(nodes[subtree.arrayIdx], undefined);
+      setDisplayOrderRecursively(nodes[subtree.arrayIdx], getDisplayOrder, undefined);
     } else {
-      yCounter = setDisplayOrderRecursively(nodes[subtree.arrayIdx], yCounter);
+      yCounter = setDisplayOrderRecursively(nodes[subtree.arrayIdx], getDisplayOrder, yCounter);
       yCounter+=spaceBetweenSubtrees;
     }
   }
   /* note that nodes[0] is a dummy node holding each subtree */
   nodes[0].displayOrder = undefined;
   nodes[0].displayOrderRange = [undefined, undefined];
+
+  timerEnd("setDisplayOrder");
 };
 
+/**
+ * @param {Array<PhyloNode>} nodes
+ * @param {boolean} focus
+ * @returns fn to return a display order (y position) for a node
+ */
+function getDisplayOrderCallback(nodes, focus) {
+  /**
+   * Start at 0 and increase with each node.
+   * Note that this value is shared across invocations of the callback.
+   */
+  let displayOrder = 0;
+
+  /**
+   * Keep track of whether the previous node was selected
+   */
+  let previousWasVisible;
+
+  if (focus) {
+    const numVisible = nodes.filter((d) => !d.hasChildren && d.visibility === NODE_VISIBLE).length;
+    const yPerFocused = (0.8 * nodes.length) / numVisible;
+    const yPerUnfocused = (0.2 * nodes.length) / (nodes.length - numVisible);
```

Comment:
This calculation translates to 80% space on focused nodes + 20% space for unfocused nodes.

The current description of the focus feature says "nodes that do not match the filter will occupy less vertical space on the tree". With the current implementation, there is a visual bug in the case of a large (>80%) selection: selected nodes will appear to shrink which is counter-intuitive.

https://github.com/user-attachments/assets/4be417e0-7fe8-47b7-84e9-6642ef5ad4d4

I'll look into adjusting the calculation to not be a fixed 80/20 and instead consider `numVisible` and `nodes.length`.

Reply:
Fixed: https://github.com/nextstrain/auspice/blob/72008c5a9d6935e69e761e3dd5eb01264d9cbd15/src/components/tree/phyloTree/helpers.js#L129-L133